src/software/parameter/parameter.h

```diff
@@ -103,32 +68,24 @@ class Parameter
      *
      * @param callback The function to call when this parameter's value is changed
      */
-    void registerCallbackFunction(std::function<void(T)> callback)
+    void registerCallbackFunction(std::function<void(T)> callback) const
     {
         std::scoped_lock callback_lock(this->callback_mutex_);
         callback_functions.emplace_back(callback);
     }
 
-   private:
-    // this mutex is marked as "mutable" so that it can be acquired in a const function
+   protected:
+    // mutexes are marked as mutable so that they can be acquired in a const function
     mutable std::mutex value_mutex_;
-    std::mutex callback_mutex_;
+    mutable std::mutex callback_mutex_;
```

Comment:
Seems like the mutexes aren't used by the derived classes so can be made `private`

Reply:
good catch, done